sequence_processing_pipeline/Job.py

```diff
@@ -290,3 +292,26 @@ def qsub(self, script_path, qsub_parameters=None,
         else:
             # job was never in the queue - return an error.
             raise PipelineError("job %s never appeared in the queue." % job_id)
+
+    def _group_commands(self, cmds):
+        # break list of commands into chunks of max_array_length (Typically
+        # 1000 for Torque job arrays). To ensure job arrays are never more
+        # than 1000 jobs long, we'll chain additional commands together, and
+        # evenly distribute them amongst the first 1000.
+        cmds.sort()
+        chunks = [cmds[i:i + self.max_array_length] for i in
+                  range(0, len(cmds), self.max_array_length)]
+
+        results = []
+
+        # create a chained command by taking one command from each list.
+        # zip_longest() allows us to handle lists of different lengths, as the
+        # last chunk will always be of different length than 1000.
+        for tuple in zip_longest(*chunks):
```

Comment:
In my tests, this does _not_ work as expected:
```python
>>> x = [1, 2, 3, 4 ,5, 6, 7]
>>> chuncks = [x[i:i + 2] for i in range(0, len(x), 2)]
>>> chuncks
[[1, 2], [3, 4], [5, 6], [7]]
>>> for tuple in zip_longest(*chunks):
...     chained_cmd = [x for x in list(tuple) if x is not None]
...     print (chained_cmd)
... 
[1, 3, 5]
[2, 4, 6]
```
see how 7 is missing.

Reply:
Can you add a tests that simulates this?

Reply:
>>> from itertools import zip_longest
>>> x = [1, 2, 3, 4 ,5, 6, 7]
>>> chunks = [x[i:i + 2] for i in range(0, len(x), 2)]
>>> chunks
[[1, 2], [3, 4], [5, 6], [7]]
>>> for tuple in zip_longest(*chunks):
...     chained_cmd = [x for x in list(tuple) if x is not None]
...     print(chained_cmd)
... 
[1, 3, 5, 7]
[2, 4, 6]
>>> 
In your example, it looks like the chunks was misspelled as 'chuncks' so I'm wondering if you had chunks declared somewhere else that the for loop would still work correctly,  but chunks was defined differently. When I try the above code with the zip_longest import added, I get the desired result.

Reply:
Sure! I'll add a test to test this method specifically.

Reply:
Thank you for confirming and yes, it works as expected.